third_party/move/move-model/bytecode/src/borrow_analysis.rs

```diff
@@ -24,13 +25,15 @@ use move_model::{
 };
 use std::{borrow::BorrowMut, collections::BTreeMap, fmt};
 
-#[derive(Debug, Clone, Eq, Ord, PartialEq, PartialOrd, Default)]
+#[derive(AbstractDomain, Debug, Clone, Eq, Ord, PartialEq, PartialOrd, Default)]
 pub struct BorrowInfo {
     /// Contains the nodes which are alive. This excludes nodes which are alive because
     /// other nodes which are alive borrow from them.
+    #[join]
     live_nodes: SetDomain<BorrowNode>,
 
     /// Forward borrow information.
+    #[join]
     borrowed_by: MapDomain<BorrowNode, SetDomain<(BorrowNode, BorrowEdge)>>,
 
     /// Backward borrow information. This field is not used during analysis, but computed once
```

Comment:
Would be good to document why a field is ignored from joins and why that is okay.

Reply:
+1